rfc/rfc-0000-k8s-events.md

```diff
@@ -0,0 +1,176 @@
+references between them. By design, the passing of references (and desired state definitions in the **owner** spec)
+is _by value_. A user can only see if a state change is transferred from source to sink, not when it happened, or how
+often, or even what the specific cause was for a resource to be updated.
+
+This approach lends itself well to the declarative, eventual consistency model of Kubernetes, where "I want this, I
+don't care how you get me there" reigns. However, there are times when a user needs to know more detail about the
+machinations of the Cartographer controller, especially in debugging failures. For these situations, Kubernetes
+provides [events](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#events)
+to expose the user to these temporal triggers/actions from inside controller processes.
+
+**Stamping** (creation of the resource definition from template and inputs) and **Submitting** (applying the resource
+definition to the cluster) are two very common events, with associated causes, that currently need further exposure to
+users.
+
+# What it is
+
+[what-it-is]: #what-it-is
+
+This first diagram shows the objects tracked or managed by the Cartographer controller, except for **Artifact** and
+**Controller Stamp**.
+
+![Cartographer Objects](./rfc-k8s-events/objects.jpg)
+
+## Developer View
+
+To date, Cartographer presents at least some information from all these sources to developers:
+
+![Owner](./rfc-k8s-events/owner.jpg)
+
+Note: `status.resources[.conditions]` do not currently look like this, but there is work to present a **Healthy** status
+[in this rfc](https://github.com/vmware-tanzu/cartographer/pull/738). Although there is no art to tackle sharing
+per-resource 'Submitted' status (I will raise a separate RFC), it's something that can be handled entirely using the
+status in a workload (ie, it does not require the use of events)
+
+With different initiatives (in RFCs for workload status improvements) the developer's view of information is becoming
+quite complete, but a key issue is the absence of awareness that "something is happening", especially after they've
+committed new code. With event's we can provide information that show's activity and causality not present in the
+current snapshot of state.
+
+We add events to the Owners (Workload/Deliverable) to enable debug and reasoning about Cartographer's behavior.
+
+## Operator View
+
+The situation is worse for users managing the supply chains (Devops/Ops). By looking at a supply chain, there is very
+little information about how supply chains are behaving, or if any issues are occurring for developers:
+
+![Blueprint](./rfc-k8s-events/blueprint.jpg)
+
+Although these users can access the logs, we can provide them a more concise view of how supply chains are behaving
+on-cluster.
+
+We add events to the Blueprints (Supply Chain/Delivery) to enable debug and reasoning about Cartographer's behavior.
```

Comment:
Won't this be a lot of events on the blueprint? Potentially thousands (or tens of thousands) Is this helpful?

Reply:
What's the concern? Calls to the API, or do you perceive there being thousands of distinct events? (recall that the server aggregates like events into a single event object with a count). 

As far as API calls go, I defer to someone like @scothis who might know what the compromise is for events.